frontend/src/component/PublicProfile/CustomFormManager.jsx

```diff
@@ -0,0 +1,289 @@
+        key={props.schemaObject.title}
+      />
+    )
+    : null
+
+  return [
+    title,
+    Object.entries(props.dataSchemaObject).map(([key, value], i) => {
+      const valueType = typeof value
+
+      if (Array.isArray(value)) {
+        return (
+          <DisplaySchemaArray
+            label={props.schemaObject.properties[key].title}
+            valueList={value}
+            currentKey={key}
+            schemaObject={props.schemaObject.properties[key].items}
+            key={`array_${key}_${i}`}
+          />
+        )
+      }
+
+      if (valueType === 'object') {
+        return (
+          <DisplaySchemaObject
+            schemaObject={props.schemaObject.properties[key]}
+            dataSchemaObject={value}
+            key={`object_${key}_${i}`}
+          />
+        )
+      }
+
+      if (valueType === 'string') {
+        return (
+          <DisplaySchemaPropertyString
+            label={props.schemaObject.properties[key].title}
+            value={value}
+            key={`property_string_${key}_${value}_${i}`}
+          />
+        )
+      }
+
+      if (valueType === 'boolean') {
+        return (
+          <DisplaySchemaPropertyBoolean
+            label={props.schemaObject.properties[key].title}
+            value={value}
+            key={`property_boolean_${key}_${value}_${i}`}
+          />
+        )
+      }
+
+      return null
+    })
+  ]
+}
+
+const SchemaAsView = props => {
+  return (
+    <div className='SchemaAsView'>
+      <DisplaySchemaObject
+        schemaObject={props.schemaObject}
+        dataSchemaObject={props.dataSchemaObject}
+      />
+
+      {props.displayEditButton && (
+        <IconButton
+          customClass={props.submitButtonClass}
+          icon='pencil-square-o'
+          onClick={props.onClickToggleButton}
+          text={props.validateLabel}
+        />
+      )}
+    </div>
+  )
+}
+
+const SchemaAsForm = props => {
+  return (
+    <Form
+      schema={props.schemaObject}
+      uiSchema={props.uiSchemaObject}
+      formData={props.dataSchemaObject}
+      onSubmit={(formData, e) => {
+        props.onClickToggleButton()
+        props.onSubmitDataSchema(formData, e)
+      }}
+    >
+      <IconButton
+        customClass={props.submitButtonClass}
+        icon='check'
+        type='submit'
+        text={props.validateLabel}
+      />
+    </Form>
+  )
+}
+
+const MODE = {
+  VIEW: 'view',
+  EDIT: 'edit'
+}
+
+export class CustomFormManager extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      mode: MODE.VIEW
+    }
+  }
+
+  handleClickToggleMode = () => {
+    const { state } = this
+    if (state.mode === MODE.VIEW) {
+      this.setState({ mode: MODE.EDIT })
+      return
+    }
+
+    if (state.mode === MODE.EDIT) {
+      this.setState({ mode: MODE.VIEW })
+    }
+  }
+
+  render () {
+    const { props, state } = this
+
+    return (
+      <div style={{ position: 'relative' }}>
```

Comment:
Why not have a class and put this style in stylus?

I think we should keep our usage of inline style to a minimum

Reply:
this was wip and is fixed